iothub_client/samples/iothub_convenience_sample/iothub_convenience_sample.c

```diff
@@ -170,7 +236,10 @@ int main(void)
     {
         // Setting message callback to get C2D messages
         (void)IoTHubDeviceClient_SetMessageCallback(device_handle, receive_msg_callback, NULL);
-        // Setting connection status callback to get indication of connection to iothub
+		// Setting method callback to handle a SetTelemetryInterval method to control
+		//   how often telemetry messages are sent from the simulated device.
+		(void)IoTHubDeviceClient_SetDeviceMethodCallback(device_handle, device_method_callback, NULL);
+		// Setting connection status callback to get indication of connection to iothub
         (void)IoTHubDeviceClient_SetConnectionStatusCallback(device_handle, connection_status_callback, NULL);
```

Comment:
@yzhong94 @ewertons 
Shouldn't we wait before using this `device_handle` until `connection_status_callback` is called with `IOTHUB_CLIENT_CONNECTION_AUTHENTICATED` ?  Currently the callback comes during the simulated device's telemetry messages being sent.